config/configunmarshaler/defaultunmarshaler.go

```diff
@@ -257,7 +258,7 @@ func unmarshalExporters(exps map[config.ComponentID]map[string]interface{}, fact
 		// Find exporter factory based on "type" that we read from config source.
 		factory := factories[id.Type()]
 		if factory == nil {
-			return nil, errorUnknownType(exportersKeyName, id)
+			return nil, errorUnknownType(exportersKeyName, id, factories)
```

Comment:
What about calling with reflect.ValueOf(factories).MapKeys() and make the error func accept []Value